### DISCUSSION TITLE
Using sampler info with $ in LGCP exposure (E) terms

### DISCUSSION THREAD

**Discussion by andyhouldcroft** [inlabru-org/inlabru#232] [General]
Hi all, 

I am interested in the functionality of E terms in LGCPs with inlabru, for example whether it is possible to include an offset for say, sampling effort, that is specific to each sampler object and is therefore geo-referenced?

I am able to run a test LGCP with e.g. "E = samplers$days" where days is the survey duration, and the lambda outputs and parameter estimates appear to be reasonable, but i am still unsure whether the days offset is actually being applied to each sampler object in space, or if it is acting in a more uniform way?

Cheers,
Andrew

**Comment by finnlindgren** [comment]
The way we implemented this initially is that the user would include a column called "weight" in the samplers data frame, so that each e.g. line transect would use the corresponding value as its own sampling effort. This should still work, as the `fm_int()` help text indicates:
 * `samplers`
     For single domain fm_int methods, an object specifying one or more subsets of the domain,
     and optional weighting in a `weight` variable.

Another option is to include this information in the predictor expression itself, as `log(effort)`, but then `effort` needs to be evaluable based on the LGCP integration points and observed point, either as a spatial variable or function call, just as how spatial covariate inputs are defined; `effort(something_that_evaluates_to_effort_values, model = "const")`

**Reply by andyhouldcroft** [reply]
that makes sense, just to clarify is the "weight" variable automatically integrated by inlabru or is E = samplers$weight still required, i.e. can a different name to "weight" be defined by the user instead?

thanks Finn!

**Reply by finnlindgren** [reply]
When using `E` with `model = "cp"`, only a common scalar is allowed. This check is in the code:
```r
    if (length(E) > 1) {
      warning("Exposure/effort parameter E should be a scalar for likelihood 'cp'.")
    }
```
So you can't use `E` for sampler-specific effort. That's what the `weight` information is for. There is currently no way to separately specify a different column to be used for `weight`; it has to be called `weight` and be in the `samplers` object. The information in it is combined with the integration weights (in the `weight` column of the output of `fm_int()`, and hence taken into account in the integration scheme.

**Reply by andyhouldcroft** [reply]
thats really helpful, thanks Finn
